MIGRATION.md

```diff
@@ -0,0 +1,193 @@
+# Migration Guide
+
+The goal of xrpl.js 3.0 was to improve the libraries support for non nodejs projects by eliminating the need for platform/build system specific polyfills and se.  This was accomplished 
+
+There were five main goals for xrpl.js 3.0 package suite:
+
+- Remove polyfills as dependencies for non node.js projects
+- Reduce the bundle size for client side applications
+- Execute unit tests in a browser environment
+- Few breaking changes
+- Improve docs
+
+The result of which is a 60% reduction in bundle size and a complete elimination of polyfills for browser applications as well as a streamlining of react-native support.  All sub packages now run their tests in both nodejs and Chrome. The `xrpl` still will not run its unit tests in the browser but it still has exhaustive coverage through its integration tests which do.
+
+## Polyfill Reduction
+
+The polyfill elimination was achieved by switching from elliptic, create-hash, and crypto polyfills to the `@noble` suite of packages which includes `@noble/hashes`, `@noble/curves`, `@scure/bip32`, @scure/bip39, and `@scure/base`.  In order to still use node's built in crypto functions when running in a node environment `@xrplf/isomorphic` was created as a way to abstract the selection of the underlying implementation.
+
+Additionally the faucet code switched from using node's `http` packages and instead leverage native fetch in the browser and `node-fetch` in node. This eliminated the polyfills for `http`, `https`, and `url`. The need for `assert` was addressed by switching to conditionals that throw exceptions and `utils` by updating error handling to use `JSON.stringify` instead of `inspect`.
+
+The removal of `buffer` proved to be a tricky endeavor.  It brought along with it the largest amount of breaking changes but was necessary to get to a polyfill free future.  Since the `Buffer` object is not native to the browser all apis were migrated to `Uint8Arrays`.  Many libraries have begun making this transition as well so with the goal in sight it made since to address it at this time. Checkout this [blogpost](https://sindresorhus.com/blog/goodbye-nodejs-buffer) by Sindre Sorhus on the rational behind using `UintArrays` over `Buffer`.
```

Comment:
nit:
```suggestion
The removal of `buffer` proved to be a tricky endeavor.  It brought along with it the largest amount of breaking changes but was necessary to get to a polyfill free future.  Since the `Buffer` object is not native to the browser all apis were migrated to `Uint8Arrays`.  Many libraries have begun making this transition as well so with the goal in sight it made since to address it at this time. Checkout this [blogpost](https://sindresorhus.com/blog/goodbye-nodejs-buffer) by Sindre Sorhus on the rationale behind using `UintArrays` over `Buffer`.
```